src/NuGetPackageVerifier/Rules/AssemblyHasAttributeRuleBase.cs

```diff
@@ -57,5 +64,11 @@ public virtual IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext
             string currentFilePath,
             AssemblyDefinition assembly,
             Mono.Collections.Generic.Collection<CustomAttribute> assemblyAttributes);
+
+        public abstract IEnumerable<PackageVerifierIssue> ValidateAttribute(
```

Comment:
This can be avoided. If `IPackageMetadata` is really needed by the sub classes, you can just add it to the existing `ValidateAttribute` abstract method. But I wouldn't really recommend it unless there are multiple rules that would benefit from this.